pkg/share/manager/json/json.go

```diff
@@ -393,7 +398,29 @@ func (m *mgr) ListReceivedShares(ctx context.Context, filters []*collaboration.F
 			rss = append(rss, rs)
 		}
 	}
-	return rss, nil
+
+	// if there is a mix-up of shares of type group and shares of type user we need to deduplicate them, since it points
```

Comment:
why is a user share more explicit than a group share? because it is a 1:1 relationship?

Reply:
I would think so. But maybe @butonic had something else in mind?